src/main/java/org/truffleruby/core/array/ArrayBuilderNode.java

```diff
@@ -232,8 +232,8 @@ public Object appendSameStrategy(Object array, int index, DynamicObject other,
             return mirror.getArray();
         }
 
-        @Specialization(guards = { "arrayStrategy != generalized",
-                                   "arrayStrategy.matchesStore(array)", "otherStrategy.matches(other)" },
+        @Specialization(guards = { "arrayStrategy.matchesStore(array)", "otherStrategy.matches(other)",
+                                   "arrayStrategy != generalized" },
```

Comment:
Maybe it doesn't matter much, but shouldn't the cheaper checks be first to help interpreter performance?

Reply:
I prefer clear complementary specializations unless there is a good incentive.
Anyway, in this case it's a test on a compilation final value and a Cached parameter, so it's only tested when creating the specialization (and `assert`-ed on the fast-path).

Reply:
It's mostly the interpreter case I was worried about. I don't want to make that unnecessarily slower if it can be easily avoided. If not applicable here, that's fine.